compiler/hie-core/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -220,37 +129,31 @@ setStageDiagnostics ::
   -- ^ the time that the file these diagnostics originate from was last edited
   stage ->
   [LSP.Diagnostic] ->
-  ProjectDiagnostics stage ->
-  ProjectDiagnostics stage
-setStageDiagnostics fp timeM stage diags (ProjectDiagnostics ds) =
-    ProjectDiagnostics $ updateDiagnostics ds uri timeM diagsBySource
+  DiagnosticStore ->
```

Comment:
The type signature looks somewhat odd now since `stage` is not tied to anything. Maybe change it to accept a `Text` and have the caller take care of the conversion?

Reply:
Agreed, changed.